api/metrics-api-jaxrs/src/main/java/org/hawkular/metrics/api/jaxrs/AvailabilityHandler.java

```diff
@@ -0,0 +1,335 @@
+
+import com.google.common.util.concurrent.Futures;
+import com.google.common.util.concurrent.ListenableFuture;
+import com.wordnik.swagger.annotations.Api;
+import com.wordnik.swagger.annotations.ApiOperation;
+import com.wordnik.swagger.annotations.ApiParam;
+import com.wordnik.swagger.annotations.ApiResponse;
+import com.wordnik.swagger.annotations.ApiResponses;
+
+/**
+ * @author Stefan Negrea
+ *
+ */
+@Path("/")
+@Consumes(APPLICATION_JSON)
+@Produces(APPLICATION_JSON)
+@Api(value = "", description = "Availability metrics interface")
+public class AvailabilityHandler {
+    private static final long EIGHT_HOURS = MILLISECONDS.convert(8, HOURS);
+
+    @Inject
+    private MetricsService metricsService;
+
+    @POST
+    @Path("/{tenantId}/availability")
+    @ApiOperation(value = "Create availability metric definition. Same notes as creating numeric metric apply.")
+    @ApiResponses(value = {
+            @ApiResponse(code = 201, message = "Metric definition created successfully"),
+            @ApiResponse(code = 400, message = "Missing or invalid payload", response = ApiError.class),
+            @ApiResponse(code = 409, message = "Availability metric with given id already exists",
+                response = ApiError.class),
+            @ApiResponse(code = 500, message = "Metric definition creation failed due to an unexpected error",
+                response = ApiError.class) })
+    public void createAvailabilityMetric(@Suspended final AsyncResponse asyncResponse,
+            @PathParam("tenantId") String tenantId, @ApiParam(required = true) AvailabilityMetric metric,
+            @Context UriInfo uriInfo) {
+        if (metric == null) {
+            Response response = Response.status(Status.BAD_REQUEST).entity(new ApiError("Payload is empty")).build();
+            asyncResponse.resume(response);
+            return;
+        }
+        metric.setTenantId(tenantId);
+        ListenableFuture<Void> future = metricsService.createMetric(metric);
+        URI created = uriInfo.getBaseUriBuilder().path("/{tenantId}/availability/{id}")
+                .build(tenantId, metric.getId().getName());
+        MetricCreatedCallback metricCreatedCallback = new MetricCreatedCallback(asyncResponse, created);
+        Futures.addCallback(future, metricCreatedCallback);
+    }
+
+    @GET
+    @Path("/{tenantId}/availability/{id}/tags")
+    @ApiOperation(value = "Retrieve tags associated with the metric definition.", response = Map.class)
+    @ApiResponses(value = {
+            @ApiResponse(code = 200, message = "Metric's tags were successfully retrieved."),
+            @ApiResponse(code = 204, message = "Query was successful, but no metrics were found."),
+            @ApiResponse(code = 500, message = "Unexpected error occurred while fetching metric's tags.",
+                response = ApiError.class) })
+    public void getAvailabilityMetricTags(@Suspended final AsyncResponse asyncResponse,
```

Comment:
There are at least a few things that fall into the category of metric meta data.

* tags
* data retention settings
* aggregation settings

Having separate endpoints that get only tags, only data retention settings, and only aggregation settings seems like overkill to me. I think something like either `GET /{tenantId}/availability/{id}/meta` or `GET /{tenantId}/availability/{id}` makes sense. I do think though that there should be different endpoints for updating each of the different meta data. Updating data retention will be an expensive, background operation. Updating aggregation settings might be as well. Updating tags is a simple map update.